homeassistant/components/tradfri/cover.py

```diff
@@ -26,9 +26,9 @@ async def async_setup_entry(
     api = tradfri_data[KEY_API]
     devices = tradfri_data[DEVICES]
 
-    covers = [dev for dev in devices if dev.has_blind_control]
-    if covers:
-        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+    async_add_entities(
+        [TradfriCover(dev, api, gateway_id) for dev in devices if dev.has_blind_control]
```

Comment:
We can make it even simpler by passing a generator expression.
```suggestion
        TradfriCover(dev, api, gateway_id) for dev in devices if dev.has_blind_control
```

Reply:
How nice, I wonder why it works. I have had problems with generator statements before where the result was either TradfriCover (if single) or [TradfriCover…]

Changed.